pkg/controller/user-cluster-controller-manager/application-installation-controller/controller.go

```diff
@@ -60,6 +62,9 @@ const (
 
 	// maxRetries is the maximum number of retries on installation or upgrade failure.
 	maxRetries = 5
+
+	// initialRequeueDuration is the time interval which is used until a node object to schedule workloads is registered in the cluster.
+	initialRequeueDuration = 15 * time.Second
```

Comment:
A bit nit-picky, but I think 10 would be the sweet spot, we can be a bit more aggressive in re-checking for node availability here.